core/server/middleware/error-handler.js

```diff
@@ -0,0 +1,109 @@
+var _ = require('lodash'),
+    path = require('path'),
+    hbs = require('express-hbs'),
+    config = require('../config'),
+    i18n = require('../i18n'),
+    _private = {};
+
+_private.parseStack = function (stack) {
+    if (!_.isString(stack)) {
+        return stack;
+    }
+
+    // TODO: split out line numbers
+    var stackRegex = /\s*at\s*(\w+)?\s*\(([^\)]+)\)\s*/i;
+
+    return (
+        stack
+            .split(/[\r\n]+/)
+            .slice(1)
+            .map(function (line) {
+                var parts = line.match(stackRegex);
+                if (!parts) {
+                    return null;
+                }
+
+                return {
+                    function: parts[1],
+                    at: parts[2]
+                };
+            })
+            .filter(function (line) {
+                return !!line;
+            })
+    );
+};
+
+_private.handleHTMLResponse = function handleHTMLResponse(err, req, res) {
+    return function handleHTMLResponse() {
+        var availableTheme = config.get('paths').availableThemes[req.app.get('activeTheme')] || {},
+            defaultTemplate = availableTheme['error.hbs'] ||
+                path.resolve(config.get('paths').adminViews, 'user-error.hbs') ||
+                'error';
+
+        res.render(defaultTemplate, {
+            message: err.message,
+            code: err.statusCode,
+            stack: _private.parseStack(err.stack)
+        }, function renderResponse(err, html) {
+            if (!err) {
+                return res.send(html);
+            }
+
+            // And then try to explain things to the user...
+            // Cheat and output the error using handlebars escapeExpression
+            return res.status(500).send(
+                '<h1>' + i18n.t('errors.errors.oopsErrorTemplateHasError') + '</h1>' +
+                '<p>' + i18n.t('errors.errors.encounteredError') + '</p>' +
+                '<pre>' + hbs.handlebars.Utils.escapeExpression(err.message || err) + '</pre>' +
+                '<br ><p>' + i18n.t('errors.errors.whilstTryingToRender') + '</p>' +
+                err.statusCode + ' ' + '<pre>' + hbs.handlebars.Utils.escapeExpression(err.message || err) + '</pre>'
+            );
+        });
+    };
+};
+
+/**
+ * @TODO: jsonapi errors format
+ */
+_private.handleJSONResponse = function handleJSONResponse(err, req, res) {
+    return function handleJSONResponse() {
+        res.json({
+            errors: [{
+                message: err.message,
+                errorType: err.errorType
```

Comment:
`TODO`: return `errorDetails` and look if something else is missing for Ghost-Admin